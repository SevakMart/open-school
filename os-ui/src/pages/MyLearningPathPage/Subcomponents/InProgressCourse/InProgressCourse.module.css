.mainContainer {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: none;
  background-color: #ffffff;
  border-radius: 20px;
  margin: 0 88px;
}
.mainContainer .titleContainer {
  margin: 32px 32px 32px 48px;
  box-sizing: border-box;
}
.mainContainer .titleContainer .courseTitle {
  font-family: "Helvetica Neue";
  font-style: normal;
  font-weight: 700;
  font-size: 24px;
  line-height: 29px;
  text-decoration-line: underline;
  color: #222427;
  margin: 0;
}
.mainContainer .titleContainer .progressContainer {
  margin: 25px 0 9px 0;
}
.mainContainer .titleContainer .progressContainer .progressBar {
  background-color: #EEEFF4;
  border-radius: 6.5px;
  width: 345px;
  height: 13px;
  margin-right: 16px;
  box-sizing: border-box;
  display: inline-block;
}
.mainContainer .titleContainer .progressContainer .progressBar .progressing {
  background-color: #4BCEB5;
  height: 100%;
  border-radius: 6.5px;
}
.mainContainer .titleContainer .progressContainer .progressPercent {
  font-family: "Helvetica Neue";
  font-style: normal;
  font-weight: 500;
  font-size: 12px;
  line-height: 15px;
  color: #333941;
  display: inline;
}
.mainContainer .titleContainer .OverallProgress {
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 400;
  font-size: 12px;
  line-height: 15px;
  color: #5E617B;
  margin: 0;
}
.mainContainer .container {
  margin: 43px;
}
.mainContainer .container .containerTitle {
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 17px;
  color: #848A9D;
  margin: 0 0 8px;
  box-sizing: border-box;
}
.mainContainer .container .remainingTimeContainer, .mainContainer .container .dueDateContainer {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: none;
}
.mainContainer .container .dueDateContent, .mainContainer .container .remainingTimeContent {
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;
  color: #333941;
  margin: 0 0 0 10px;
  display: inline;
}
.mainContainer .container .statusContent {
  background-color: #D9DBE9;
  border-radius: 16px;
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 500;
  font-size: 12px;
  line-height: 15px;
  color: #5E617B;
  padding: 0.344rem 1rem 0.344rem 1rem;
  margin: 8px 0 0;
}
.mainContainer .buttonContainer {
  margin-left: 43px;
}
.mainContainer .separator {
  background-color: #EEEFF4;
  border-radius: 0.469rem;
  width: 0.125rem;
  height: 5rem;
}/*# sourceMappingURL=InProgressCourse.module.css.map */