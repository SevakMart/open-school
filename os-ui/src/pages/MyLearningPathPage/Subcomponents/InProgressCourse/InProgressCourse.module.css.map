{"version":3,"sources":["InProgressCourse.module.scss","../../../../styles/Mixins/FlexMixins.scss","../../../../styles/Configs/Colors.scss","InProgressCourse.module.css","../../../../styles/Mixins/FontMixin.scss","../../../../styles/Configs/Fonts.scss"],"names":[],"mappings":"AAMA;ECLE,aAAA;EACA,mBDKqB;ECJrB,mBDIyB;ECHzB,qBDGgC;EAC9B,yBEJe;EFKf,mBAAA;EACA,cAAA;AGFJ;AHGI;EACI,2BAAA;EACA,sBAAA;AGDR;AHEQ;EIbN,6BAAA;EACA,kBJa6C;EIZ7C,gBJYoD;EIXpD,eJWwD;EIVxD,iBJU6D;EACnD,+BAAA;EACA,cEdQ;EFeR,SAAA;AGIZ;AHFQ;EACI,oBAAA;AGIZ;AHHW;EACG,yBAAA;EACA,oBAAA;EACA,YAAA;EACA,YAAA;EACA,kBAAA;EACA,sBAAA;EACA,qBAAA;AGKd;AHJc;EACE,yBE9BE;EF+BF,YAAA;EACA,oBAAA;AGMhB;AHHY;EInCV,6BAAA;EACA,kBJmC+C;EIlC/C,gBJkCsD;EIjCtD,eJiC0D;EIhC1D,iBJgC+D;EACnD,cEjCK;EFkCL,eAAA;AGSd;AHNY;EIzCV,yBCDY;EDEZ,kBJyC6C;EIxC7C,gBJwCoD;EIvCpD,eJuCwD;EItCxD,iBJsC6D;EACjD,cE1CI;EF2CJ,SAAA;AGYd;AHTI;EACI,YAAA;AGWR;AHVO;EIjDL,yBCDY;EDEZ,kBJiDyC;EIhDzC,gBJgDgD;EI/ChD,eJ+CoD;EI9CpD,iBJ8CyD;EACjD,cAAA;EACA,eAAA;EACA,sBAAA;AGgBV;AHdO;ECvDL,aAAA;EACA,mBDuD2B;ECtD3B,mBDsD+B;ECrD/B,qBDqDsC;AGmBxC;AHjBO;EI1DL,yBCDY;EDEZ,kBJ0DuC;EIzDvC,gBJyD8C;EIxD9C,eJwDkD;EIvDlD,iBJuDuD;EACjD,cExDW;EFyDX,kBAAA;EACA,eAAA;AGuBR;AHpBO;EACI,yBAAA;EACA,mBAAA;EInET,yBCDY;EDEZ,kBJmE0C;EIlE1C,gBJkEiD;EIjEjD,eJiEqD;EIhErD,iBJgE0D;EACjD,cEpEO;EFqEP,oCAAA;EACA,eAAA;AG0BX;AHvBI;EACI,iBAAA;AGyBR;AHvBI;EACI,yBAAA;EACA,uBAAA;EACA,eAAA;EACA,YAAA;AGyBR","file":"InProgressCourse.module.css"}