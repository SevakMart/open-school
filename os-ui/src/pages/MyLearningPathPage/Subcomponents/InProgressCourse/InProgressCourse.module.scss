@import "../../../../styles/Mixins/ButtonMixin.scss";
@import "../../../../styles/Mixins/FlexMixins.scss";
@import "../../../../styles/Mixins/FontMixin.scss";

.mainContainer{
    @include flexStyle(row,center,space-between);
    background-color: #FFFFFF;
    border-radius:1.25rem;
    width:100%;
    margin-bottom:7%;
    padding-left:3rem;
    padding-right:3rem;
    .container{
        @include flexStyle(column,center,space-between);
        .courseTitle{
            @include fontStyle('Helvetica Neue',normal,700,1.5rem,1.813rem);
            text-decoration-line: underline;
            color: #222427;
        }
        .progressContainer{
            @include flexStyle(row,center,space-between);
            width:80%;
        .progressBar{
            background-color: #EEEFF4;
            border-radius:0.406rem;
            width:100%;
            height:0.813rem;
        }
        .progressPercent{
            @include fontStyle('Helvetica Neue',normal,500,0.75rem,0.938rem);
            color: #333941;
        }
      }
        .OverallProgress{
            @include fontStyle('Montserrat',normal,400,0.75rem,0.938rem);
            color: #5E617B;
       }
       .containerTitle{
        @include fontStyle('Montserrat',normal,500,0.875rem,1.063rem);
        color: #848A9D;
       }
       .remainingTimeContainer,.dueDateContainer{
        @include flexStyle(row,center,space-between);
       }
       .dueDateContent,.remainingTimeContent{
        @include fontStyle('Montserrat',normal,500,1rem,1.5rem);
        color: #333941;
       }
       .statusContent{
           background-color: #D9DBE9;
           border-radius:1rem;
           @include fontStyle('Montserrat',normal,500,0.75rem,0.938rem);
           color: #5E617B;
           padding:0.344rem 1rem 0.344rem 1rem;
       }
       &:nth-child(1){
           width:25%;
           border:none;
       }
    }
    .buttonContainer button{
        background-color: #848A9D;
        border-radius:1rem;
        @include fontStyle('Montserrat',normal,500,1rem,1.25rem);
        color:#FFFFFF;
        padding:1rem 2rem 1rem 2rem;
        cursor:pointer;
    }
    .separator{
        background-color:#EEEFF4;
        border-radius: 0.469rem;
        width:0.125rem;
        height:5rem;
    }
}
