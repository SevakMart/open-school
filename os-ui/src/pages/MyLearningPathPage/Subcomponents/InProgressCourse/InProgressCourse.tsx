import { useState } from 'react';
import { UserCourseType } from '../../../../types/UserCourseType';
import TimeIcon from '../../../../icons/Time';
import CalendarIcon from '../../../../icons/CalenderIcon';
import { transformTime } from '../../../../helpers/TimeTransform';
import styles from './InProgressCourse.module.scss';

const InProgressCourse = ({
  title, courseStatus, percentage, remainingTime, dueDate,
}:Omit<UserCourseType, 'grade'>) => {
  const {
    mainContainer, container, progressBar, buttonContainer, courseTitle,
    progressPercent, OverallProgress, containerTitle, progressing, progressContainer,
    dueDateContent, remainingTimeContent, statusContent, separator,
    remainingTimeContainer, dueDateContainer,
  } = styles;
  const time = useState(transformTime(remainingTime))[0];
  return (
    <div className={mainContainer}>
      <div className={container}>
        <p className={courseTitle}>{title}</p>
        <div className={progressContainer}>
          <div className={progressBar}>
            <div
              className={progressing}
              style={{
                width: `${percentage}%`, backgroundColor: '#A0A3BD', height: '100%', borderRadius: '0.406rem',
              }}
            />
          </div>
          <p className={progressPercent}>
            {percentage}
            %
          </p>
        </div>
        <p className={OverallProgress}>Overall Progress</p>
      </div>
      <div className={separator} />
      <div className={container}>
        <p className={containerTitle}>Remaining Time</p>
        <div className={remainingTimeContainer}>
          <p><TimeIcon /></p>
          <p className={remainingTimeContent}>{time}</p>
        </div>
      </div>
      <div className={separator} />
      <div className={container}>
        <p className={containerTitle}>Status</p>
        <div className={statusContent}>{courseStatus}</div>
      </div>
      <div className={separator} />
      <div className={container}>
        <p className={containerTitle}>Due Date</p>
        <div className={dueDateContainer}>
          <p><CalendarIcon /></p>
          <p className={dueDateContent}>{dueDate}</p>
        </div>
      </div>
      <div className={separator} />
      <div className={buttonContainer}>
        <button type="button">Resume Course</button>
      </div>
    </div>
  );
};
export default InProgressCourse;
