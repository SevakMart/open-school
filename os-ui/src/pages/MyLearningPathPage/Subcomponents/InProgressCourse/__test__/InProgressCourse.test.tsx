import { render, screen } from '@testing-library/react';
import InProgressCourse from '../InProgressCourse';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key:string) => key }),
}));

describe('Create test cases for InProgress component', () => {
  test('Create a snapshot test', () => {
    const { asFragment } = render(
      <InProgressCourse
        title="React js"
        courseStatus="In_Progress"
        percentage={47}
        remainingTime={540000}
        dueDate="2022-05-07"
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test('Check if the content is as expected', () => {
    render(
      <InProgressCourse
        title="React js"
        courseStatus="In_Progress"
        percentage={47}
        remainingTime={5400000}
        dueDate="2022-05-07"
      />,
    );
    const courseTitleElement = screen.queryByTestId('React js');
    const statusContentElement = screen.queryByTestId('In_Progress');
    const percentageContentElement = screen.queryByTestId(47);
    const remainingTimeContentElement = screen.queryByTestId(5400000);
    const dueDateContentElement = screen.queryByTestId('2022-05-07');

    expect(courseTitleElement).toBeInTheDocument();
    expect(courseTitleElement).toHaveTextContent('React js');
    expect(statusContentElement).toBeInTheDocument();
    expect(statusContentElement).toHaveTextContent('In_Progress');
    expect(percentageContentElement).toBeInTheDocument();
    expect(percentageContentElement).toHaveTextContent('47%');
    expect(remainingTimeContentElement).toBeInTheDocument();
    expect(remainingTimeContentElement).toHaveTextContent('1h 30m');
    expect(dueDateContentElement).toBeInTheDocument();
    expect(dueDateContentElement).toHaveTextContent('2022-05-07');
  });
});
