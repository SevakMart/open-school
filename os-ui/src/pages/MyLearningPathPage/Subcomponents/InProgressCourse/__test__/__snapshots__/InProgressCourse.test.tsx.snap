// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create test cases for InProgress component Create a snapshot test 1`] = `
<DocumentFragment>
  <div
    class="mainContainer"
  >
    <div
      class="titleContainer"
    >
      <p
        class="courseTitle"
        data-testid="React js"
      >
        React js
      </p>
      <div
        class="progressContainer"
      >
        <div
          class="progressBar"
        >
          <div
            class="progressing"
            style="width: 47%;"
          />
        </div>
        <p
          class="progressPercent"
          data-testid="47"
        >
          47%
        </p>
      </div>
      <p
        class="OverallProgress"
      >
        string.myLearningPaths.inProgressCourse.overallProgress
      </p>
    </div>
    <div
      class="separator"
    />
    <div
      class="container"
    >
      <p
        class="containerTitle"
      >
        string.myLearningPaths.inProgressCourse.remainingTime
      </p>
      <div
        class="remainingTimeContainer"
      >
        <img
          alt="clockIcon"
          class="timeIcon"
          src="ClockIcon.svg"
        />
        <p
          class="remainingTimeContent"
          data-testid="90"
        >
          90 minutes
        </p>
      </div>
    </div>
    <div
      class="separator"
    />
    <div
      class="container"
    >
      <p
        class="containerTitle"
      >
        string.myLearningPaths.inProgressCourse.status
      </p>
      <p
        class="statusContent"
        data-testid="In_Progress"
      >
        In_Progress
      </p>
    </div>
    <div
      class="separator"
    />
    <div
      class="container"
    >
      <p
        class="containerTitle"
      >
        string.myLearningPaths.inProgressCourse.dueDate
      </p>
      <div
        class="dueDateContainer"
      >
        <img
          alt="calendarIcon"
          class="calenderIcon"
          src="CalendarIcon.svg"
        />
        <p
          class="dueDateContent"
          data-testid="2022-05-07"
        >
          2022-05-07
        </p>
      </div>
    </div>
    <div
      class="separator"
    />
    <div
      class="buttonContainer"
    >
      <button
        class="resumeCourse"
        type="button"
      >
        button.myLearningPathsPage.resumeCourse
      </button>
    </div>
  </div>
</DocumentFragment>
`;
